pkg/cassandra/nodetool/nodetool.go

```diff
@@ -0,0 +1,170 @@
+package nodetool
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/pborman/uuid"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	"github.com/jetstack/navigator/pkg/cassandra/nodetool/client"
+)
+
+type NodeState string
+
+const (
+	NodeStateUnknown NodeState = "Unknown"
+	NodeStateNormal  NodeState = "Normal"
+	NodeStateLeaving NodeState = "Leaving"
+	NodeStateJoining NodeState = "Joining"
+	NodeStateMoving  NodeState = "Moving"
+)
+
+type NodeStatus string
+
+const (
+	NodeStatusUnknown NodeStatus = "Unknown"
+	NodeStatusUp      NodeStatus = "Up"
+	NodeStatusDown    NodeStatus = "Down"
+)
+
+type Node struct {
+	Host   string
+	ID     uuid.UUID
+	State  NodeState
+	Status NodeStatus
+	Local  bool
+}
+
+type NodeMap map[string]*Node
+
+func (nm NodeMap) LocalNode() *Node {
+	for _, node := range nm {
+		if node.Local {
+			return node
+		}
+	}
+	return nil
+}
```

Comment:
Might be worth adjusting this method to return an error if the LocalNode isn't found? Else we might accidentally get some nil pointer exceptions further down the line.

Reply:
I'll leave it for now.